src/main/java/edu/ucdavis/dss/ipa/services/jpa/JpaCourseService.java

```diff
@@ -211,6 +211,42 @@ public Course findOrCreateBySubjectCodeAndCourseNumberAndSequencePatternAndTitle
 		return course;
 	}
 
+	@Override
+	public Course findOrCreateByCourse(Course courseDTO) {
```

Comment:
Javadoc this function to explain when a course matches versus when it doesn't, plus whatever else would be in the Javadoc.